src/main/java/com/gluonhq/RunMojo.java

```diff
@@ -112,7 +120,11 @@ public void execute() throws MojoExecutionException {
         }
 
         invocationRequest.setPomFile(runPomFile);
-        invocationRequest.setGoals(Collections.singletonList("javafx:run"));
+        if (execution == null) {
```

Comment:
Looks good, however I don't think `execution` can be null. If there is no explicit `<execution>` in the pom, it uses  `default-cli`.

For instance, when you run `mvn javafx:run` you see:
```
[INFO] --- javafx-maven-plugin:0.0.6:run (default-cli) 
``